rollup/rollupsyncservice/chunk.go

```diff
@@ -0,0 +1,155 @@
+
+		// Append rlp-encoded l2Txs
+		for _, txData := range block.Transactions {
+			if txData.Type == types.L1MessageTxType {
+				continue
+			}
+			rlpTxData, err := convertTxDataToRLPEncoding(txData)
+			if err != nil {
+				return nil, err
+			}
+			var txLen [4]byte
+			binary.BigEndian.PutUint32(txLen[:], uint32(len(rlpTxData)))
+			l2TxDataBytes = append(l2TxDataBytes, txLen[:]...)
+			l2TxDataBytes = append(l2TxDataBytes, rlpTxData...)
+		}
+	}
+
+	chunkBytes = append(chunkBytes, l2TxDataBytes...)
+
+	return chunkBytes, nil
+}
+
+// Hash hashes the Chunk into RollupV2 Chunk Hash
+func (c *Chunk) Hash(totalL1MessagePoppedBefore uint64) (common.Hash, error) {
+	chunkBytes, err := c.Encode(totalL1MessagePoppedBefore)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	numBlocks := chunkBytes[0]
+
+	// concatenate block contexts
+	var dataBytes []byte
+	for i := 0; i < int(numBlocks); i++ {
+		// only the first 58 bytes of each BlockContext are needed for the hashing process
+		dataBytes = append(dataBytes, chunkBytes[1+60*i:60*i+59]...)
+	}
+
+	// concatenate l1 and l2 tx hashes
+	for _, block := range c.Blocks {
+		var l1TxHashes []byte
+		var l2TxHashes []byte
+		for _, txData := range block.Transactions {
+			txHash := strings.TrimPrefix(txData.TxHash, "0x")
+			hashBytes, err := hex.DecodeString(txHash)
+			if err != nil {
+				return common.Hash{}, err
+			}
+			if txData.Type == types.L1MessageTxType {
+				l1TxHashes = append(l1TxHashes, hashBytes...)
+			} else {
+				l2TxHashes = append(l2TxHashes, hashBytes...)
+			}
+		}
+		dataBytes = append(dataBytes, l1TxHashes...)
+		dataBytes = append(dataBytes, l2TxHashes...)
+	}
+
+	hash := crypto.Keccak256Hash(dataBytes)
+	return hash, nil
+}
+
+// DecodeChunkBlockRanges decodes the provided chunks into a list of block ranges. Each chunk
+// contains information about multiple blocks, which are decoded and their ranges (from the
+// start block to the end block) are returned.
+func DecodeChunkBlockRanges(chunks [][]byte) ([]*rawdb.ChunkBlockRange, error) {
+	var chunkRanges []*rawdb.ChunkBlockRange
+	for _, chunk := range chunks {
+		if len(chunk) < 1 {
+			return nil, fmt.Errorf("invalid chunk, length is less than 1")
+		}
+
+		numBlocks := int(chunk[0])
+		if len(chunk) < 1+numBlocks*blockContextByteSize {
+			return nil, fmt.Errorf("chunk size doesn't match with numBlocks")
+		}
+
+		blockContexts := make([]*BlockContext, numBlocks)
+		for i := 0; i < numBlocks; i++ {
+			startIdx := 1 + i*blockContextByteSize // add 1 to skip numBlocks byte
+			endIdx := startIdx + blockContextByteSize
+			blockContext, err := decodeBlockContext(chunk[startIdx:endIdx])
+			if err != nil {
+				return nil, err
+			}
+			blockContexts[i] = blockContext
+		}
+
+		chunkRanges = append(chunkRanges, &rawdb.ChunkBlockRange{
+			StartBlockNumber: blockContexts[0].BlockNumber,
+			EndBlockNumber:   blockContexts[len(blockContexts)-1].BlockNumber,
+		})
+	}
+	return chunkRanges, nil
```

Comment:
```suggestion
		chunkBlockRanges = append(chunkBlockRanges, &rawdb.ChunkBlockRange{
			StartBlockNumber: blockContexts[0].BlockNumber,
			EndBlockNumber:   blockContexts[len(blockContexts)-1].BlockNumber,
		})
	}
	return chunkBlockRanges, nil
```

Reply:
renamed in [65629da](https://github.com/scroll-tech/go-ethereum/pull/515/commits/65629daef071314a5271b91e15672cf64dce1454)